mantidproject/mantid
# Offset in asymmetry calculations: new algorithms

The asymmetry calculations in muon analysis were not centered on zero. Two new algorithms have been made to calculate the asymmetry (an estimate and a full calculation). 

**To test:**


The easiest way to test the data is to open the muon analysis interface and load a MUSR file. Change the Plot type to counts (from asymmetry). The plot should update. 

Then run EstimateAsymmetryFromCounts and the input workspace should be MuonAnalysisGrouped_1 (there are only 4 spectra). The result should be a spectrum that oscillates around a value close to zero (unless a zero field case has been used, which should increase over time). I find that an x range of 0 to 15 and a y range of -30 to 30 works well.  

Then run CalculateAsymmetry and the input workspace should be MuonAnalysisGrouped_1. You will then be able to enter a function to use (can do a list) and the default is good for MUSR data. The result should be a spectrum that oscillates around a value of zero, test this by fitting a flatbackground and a GaussOsc to the output. This will not work for zero field. 

Is the first stage of fixing https://github.com/mantidproject/mantid/issues/18989 and the next stage will be to add these algorithms to the muon analysis gui. 



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
